lib/block/blocktransfer-error.js

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const errorCodes = require('./error-codes');
+
+/**
+ * Custom Error class containing additional information specific to the download process. This primarily consists of an
+ * error code, which can be used by consumers to provide more specific information about the nature of an error.
+ */
+class BlockTransferError extends Error {
+
+    /**
+     * Concatenates to message values together if both are provided.
+     *
+     * @param {string} overallMessage Will be prepended to specificMessage, delimited with a colon, if supplied.
+     * @param {string} specificMessage Will be concatenated with overallMessage, if supplied. 
+     *  Otherwise the return value of the method will be specificMessage as-is.
+     * @returns {string} A message value.
+     */
+    static getFullMessage(overallMessage, specificMessage) {
+        if (overallMessage) {
+            return `${overallMessage}: ${specificMessage}`;
+        }
+        return specificMessage;
+    }
+
+
+    /**
+     * Constructs a new instance containing the provided information.
+     *
+     * @param {string} message The message that will appear with the Error instance.
+     * @param {string} code The code indicating the specific type of error.
+     * @param {string} [innerStack] Additional stack information if the UploadError instance originated
+     *  from another Error.
+     */
+    constructor(message, code, innerStack = '') {
+        super(message);
+        this.code = code;
+        this.innerStack = innerStack;
+        this.uploadError = true;
+    }
+
+    /**
+     * Retrieves the error code representing the specific type of error. See ErrorCodes for more
+     * information.
+     *
+     * @returns {string} An error code value.
+     */
+    getCode() {
+        return this.code;
```

Comment:
`.code` can still be used to access the code right? then why is this needed?

Reply:
Yes, no change here, except where it's located.